yesod-core/Yesod/Core/Class/Yesod.hs

```diff
@@ -366,6 +366,22 @@ sslOnlySessions = (fmap . fmap) secureSessionCookies
     setSecureBit cookie = cookie { setCookieSecure = True }
     secureSessionCookies = customizeSessionCookies setSecureBit
 
+-- | Helps defend against CSRF attacks by setting the SameSite attribute on
+-- session cookies to "Lax".
```

Comment:
I think this documentation would be more helpful if it briefly explained what this attribute does, and/or linked to an external resource with a good explanation, similar to `sslOnlySessions`'s documentation. That would help users who are trying to figure out how Yesod protects against CSRF attacks, but aren't familiar with this specific method of prevention. What do you think of something like this?

```
-- | Helps defend against CSRF attacks by setting the SameSite attribute on
-- session cookies to "Lax". See the <https://tools.ietf.org/html/draft-west-first-party-cookies IETF Draft for details>.
```

or more detailed:

```
-- | Helps defend against CSRF attacks by setting the SameSite attribute on
-- session cookies to "Lax". This will cause browsers to not include your session cookie
-- when a write request is made from www.attacker.com to your site. HTTP requests using
-- @GET@, @HEAD@, @OPTIONS@ or @TRACE@ aren't affected by this setting.
-- 
-- See the <https://tools.ietf.org/html/draft-west-first-party-cookies IETF Draft for Details>.
-- 
-- /Warning:/ SameSession cookies were introduced mid-2016, so many browsers won't support them.
-- You should use SameSite session cookies in combination with normal CSRF prevention methods.
laxSameSiteSessions :: ...
```

```
-- | Helps defend against CSRF attacks by setting the SameSite attribute on
-- session cookies to "Strict". The "Strict" option functions the same as the "Lax" option,
-- but also removes cookies for @GET@, @HEAD@, @OPTIONS@ and @TRACE@ HTTP methods.
strictSameSiteSessions :: ...
```

Caveat: I only learned about the SameSite cookie attribute when you opened this PR, so you probably know more about what this documentation should say